lua/lean/html.lua

```diff
@@ -560,12 +560,11 @@ function Div:buf_hover()
   else
     bufdata.hover_range = nil
   end
-  if not vim.deep_equal(old_hover_range, bufdata.hover_range) then
-    local hlgroup = "htmlDivHighlight"
-    vim.api.nvim_buf_clear_namespace(bufdata.buf, hl_ns, 0, -1)
-    if bufdata.hover_range then
-      vim.highlight.range(bufdata.buf, hl_ns, hlgroup, bufdata.hover_range[1], bufdata.hover_range[2])
-    end
+
+  local hlgroup = "htmlDivHighlight"
```

Comment:
Not sure exactly what was happening, but in Lean 3 when I opened a tooltip in a non-current pin the highlights would disappear. Because clicking a widget in Lean 3 updates the other pins with that same URI, I think that the highlights from the render namespace were overriding the hover highlights for some reason.

Reply:
I'm thinking we could properly fix this by using extmarks instead of highlights for the hover namespace

Reply:
I think you're spot on with the render-clears-highlight hypothesis.  I can even reproduce this even with a single pin, opening the tooltip removes the highlighting.  I took the liberty of adding back the deep_equals, but with a flag that buf_render sets to override the check.

Reply:
I don't think this can be fixed with extmarks though.  Extmarks can track edits, but I think if you replace the whole buffer then neovim will just remove the old extmarks.